fhir_zulip_nlp/fhir_zulip_nlp.py

```diff
@@ -304,41 +343,66 @@ def query_categories(category_keywords: Dict[str, List[str]]) -> pd.DataFrame:
-def _get_keywords(
-    sheet_name='category_keywords', cache_path=CONFIG['keywords_cache_path'], use_cache_only=False
-) -> Dict[str, List[str]]:
-    """Get keywords from google sheets, else cache
-    sheet_name: The name of the specific sheet within a GoogleSheet workbook."""
-    # Load
-    if use_cache_only:
-        df: pd.DataFrame = pd.read_csv(cache_path)
-    else:
-        try:
-            # todo: Pass URI too
-            df: pd.DataFrame = get_sheets_data(sheet_name=sheet_name, env_dir=ENV_DIR)
-            df.to_csv(cache_path, index=False)
-        except:
-            df: pd.DataFrame = pd.read_csv(cache_path)
-
+def _get_keywords(use_cached_keyword_inputs=False) -> TYPE_KEYWORDS_DICT:
+    """Get keywords iterable"""
+    df: pd.DataFrame = _load_keywords_df(use_cache_only=use_cached_keyword_inputs)
     # Convert to dictionary
-    category_keywords: Dict[str, List[str]] = {}
+    category_keywords: TYPE_KEYWORDS_DICT = {}
     categories = list(df['category'].unique())
     for c in categories:
+        category_keywords[c] = {}
         df_i = df[df['category'] == c]
-        keywords = list(df_i['keyword'])
-        category_keywords[c] = keywords
+        for _index, row in df_i.iterrows():
+            kw = row['keyword']
+            spellings_str: str = row['spelling_variations']
+            spellings_list: List[str] = spellings_str.split(INTRA_CELL_DELIMITER) if spellings_str else [kw]
+            category_keywords[c][kw] = spellings_list
 
     return category_keywords
 
 
-def run():
+def _get_keyword_contexts(use_cached_keyword_inputs=False) -> Dict[str, List[str]]:
+    """Get keyword contexts
+    If keyword has no contexts, will not be included in the dictionary."""
+    df: pd.DataFrame = _load_keywords_df(use_cache_only=use_cached_keyword_inputs)
+    # Convert to dictionary
+    keyword_contexts: Dict[str, List[str]] = {}
+    for _index, row in df.iterrows():
+        kw = row['keyword']
+        context_str: str = row['context']
+        context_list: List[str] = context_str.split(INTRA_CELL_DELIMITER) if context_str else []
+        if context_list:
+            keyword_contexts[kw] = context_list
+
+    return keyword_contexts
+
+
+def run(analyze_only=False, use_cached_keyword_inputs=False):
     """Run program"""
-    keywords = _get_keywords()
-    return query_categories(keywords)
+    keywords: TYPE_KEYWORDS_DICT = _get_keywords(use_cached_keyword_inputs)
+    message_df: pd.DataFrame = query_categories(keywords) if not analyze_only else _load_cached_messages()
+    kw_contexts: Dict[str, List[str]] = _get_keyword_contexts()
+    # - report 1: counts and latest/oldest message timestamps && keywords w/ no results
+    create_report1(df=message_df, category_keywords=keywords, kw_contexts=kw_contexts)
+    # - report 2: thread lengths
+    create_report2(df=message_df, category_keywords=keywords, kw_contexts=kw_contexts)
+
+
+def cli():
```

Comment:
@rohaher Here's a new CLI. You can still run the program as you would before. But it supports these additional options now. Basically if these flags (`--analyze-only`, `--use-cahced-keyword-inputs`) are left out, they will default to `False`. If they're included, they'll be `True`. They're both boolean options. That's what the `action='store_true'` is about.